deploy/helm/sumologic/templates/logs/collector/otelcol/daemonset.yaml

```diff
@@ -1,59 +1,70 @@
 {{- if eq (include "logs.collector.otelcol.enabled" .) "true" }}
+{{- $ctx := . }}
+{{- $instance := "" -}}
+{{- $daemonsets := dict "" $.Values.otellogs.daemonset  -}}
+{{- $daemonsets = deepCopy $daemonsets | merge $.Values.otellogs.additionalDaemonSets -}}
+{{- range $name, $value := $daemonsets }}
+{{- if not (eq $name "") }}
+{{- $instance = (printf "-%s" $name ) }}
+{{- end }}
+{{- $resources := deepCopy $.Values.otellogs.daemonset.resources | merge .resources }}
+---
 apiVersion: apps/v1
 kind: DaemonSet
 metadata:
-  name: {{ template "sumologic.metadata.name.logs.collector.daemonset" . }}
+  name: {{ printf "%s%s" (include "sumologic.metadata.name.logs.collector.daemonset" $ctx) $instance }}
```

Comment:
You should truncate this to 63 chars, since it's derived from user input. Same with the labels.